app/scripts/controllers/metametrics.ts

```diff
@@ -470,7 +498,22 @@ export default class MetaMetricsController {
     const fragment = fragments[id];
 
     if (!fragment) {
-      throw new Error(`Event fragment with id ${id} does not exist.`);
+      if (!id.includes('transaction-submitted-')) {
+        throw new Error(`Event fragment with id ${id} does not exist.`);
+      }
+
+      /**
+       * HACK: "transaction-submitted-<id>" fragment hack
+       * Creates a "transaction-submitted-<id>" fragment if it does not exist to persist
+       * accumulated event metrics. In the case it is unused, the abandoned fragment will
+       * eventually be deleted with canThrowAwayIfAbandoned set to true.
+       */
+      fragments[id] = {
```

Comment:
Rather than updating the state object directly, which we then merge with below, would it be cleaner to populate a new object such as `options` that defaults to empty and we reference below in the `merge` with a spread operator?

Reply:
I think this could be a good solution as well. It wasn't considered due to time constraints, but since we are not cherry-picking the solution, I can take a wing at this. I'm still in favor of solution A, but who knows when we could work on that solution.

Reply:
For clarity, this doesn't change the core architecture of the fix, I'm just meaning we can define a local variable to pass to `merge` below rather than referencing `fragments[id]` directly.

Reply:
ah thanks for the clarification! ok, yeah, this could be done

Reply:
updated [0f4c0d9](https://github.com/MetaMask/metamask-extension/pull/28240/commits/0f4c0d9a487aa4b835ac05e34d9654d5135fc133)